src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,176 @@
+from logging import getLogger
+from yql.api.v1.client import YqlClient
+import time
+import datetime
+
+
+logger = getLogger('mm.planner')
+
+
+class YqlWrapper(object):
+    def __init__(self, cluster, token, attempts, delay):
+        self._cluster = cluster
+        self._token = token
+        self._attempts = attempts
+        self._delay = delay
+        self.aggregate_query = """
+        SELECT couple_id
+        FROM
+            (SELECT
+                        couple_id,
+                        SUM(expired_size) AS sum_expired_size
+                    FROM [{table}]
+                    WHERE expiration_date <= {timestamp}
+                    GROUP BY couple_id)
+            WHERE  sum_expired_size >= {trigger};
+        """
+
+        # XXX: maybe something more optimal could be done? How the db is organized?
+        # XXX: tounixdate -> arithmetic ops
+        # XXX: use HAVING ?
+        self.preaggregate_query = """
```

Comment:
7459417 Host: n0286-sas.hahn.yt.yandex.net:9012
7459418 Error: {"attributes"={"fid"=18446486516982214696u;"tid"=7467061416912571830u;"datetime"="2016-12-10T09:00:01.002785Z";"pid"=689961;"host"="n0286-sas-i.hahn.yt.yandex.net"};"        code"=1;"inner_errors"=[{"attributes"={"fid"=18446486516806784911u;"tid"=17803177109177937207u;"datetime"="2016-12-10T08:59:58.696003Z";"pid"=689961;"exit_code"=134;"host"="        n0286-sas-i.hahn.yt.yandex.net"};"code"=10000;"message"="Process exited with code 134"}];"message"="User job failed"}
7459419 Stderr:
7459420 Loaded UDF module: Python, from ./ff61ecc4-d093471c-d5022908-16, ABI version 1.3.0
7459421 Terminate was called, reason: Failed to compile module:   File "embedded:tounixdate", line 2
7459422     import time
7459423    ^
7459424 IndentationError: unexpected indent
7459425 
7459426 /bin/bash: line 1: 690024 Aborted                 ./cppbinary --yt-map "NYql::NAgent::TYQLYtMap<NYT::TNode>" 1 1
7459427 
7459428 Host: n4075-sas.hahn.yt.yandex.net:9012
7459429 Error: {"attributes"={"fid"=18446486516980997641u;"tid"=17742262749045998366u;"datetime"="2016-12-10T09:00:00.733057Z";"pid"=95760;"host"="n4075-sas-i.hahn.yt.yandex.net"};"        code"=1;"inner_errors"=[{"attributes"={"fid"=18446486516803133746u;"tid"=2209842740423266240u;"datetime"="2016-12-10T08:59:58.354727Z";"pid"=95760;"exit_code"=134;"host"="n4        075-sas-i.hahn.yt.yandex.net"};"code"=10000;"message"="Process exited with code 134"}];"message"="User job failed"}
7459430 Stderr:
7459431 Loaded UDF module: Python, from ./ff61ecc4-d093471c-d5022908-16, ABI version 1.3.0
7459432 Terminate was called, reason: Failed to compile module:   File "embedded:tounixdate", line 2
7459433     import time
7459434    ^
7459435 IndentationError: unexpected indent
7459436 
7459437 /bin/bash: line 1: 95867 Aborted                 ./cppbinary --yt-map "NYql::NAgent::TYQLYtMap<NYT::TNode>" 1 1
7459438 
7459439 Host: n0286-sas.hahn.yt.yandex.net:9012
7459440 Error: {"attributes"={"fid"=18446486516981519236u;"tid"=13285400733211156866u;"datetime"="2016-12-10T09:00:00.852335Z";"pid"=689966;"host"="n0286-sas-i.hahn.yt.yandex.net"};        "code"=1;"inner_errors"=[{"attributes"={"fid"=18446486516798787121u;"tid"=1799316744586959062u;"datetime"="2016-12-10T08:59:58.700245Z";"pid"=689966;"exit_code"=134;"host"="        n0286-sas-i.hahn.yt.yandex.net"};"code"=10000;"message"="Process exited with code 134"}];"message"="User job failed"}
7459441 Stderr:
7459442 Loaded UDF module: Python, from ./ff61ecc4-d093471c-d5022908-16, ABI version 1.3.0
7459443 Terminate was called, reason: Failed to compile module:   File "embedded:tounixdate", line 2
7459444     import time
7459445    ^
7459446 IndentationError: unexpected indent